adapter/adapter.go

```diff
@@ -250,7 +283,9 @@ func OnChannelMessage(ctx context.Context, event *ProviderEvent, data *ChannelMe
 	}
 
 	// Remove the "trigger character" (!)
-	rawCommandText = rawCommandText[1:]
+	if rawCommandText[0] == '!' {
+		rawCommandText = rawCommandText[1:]
+	}
```

Comment:
Should we have two logical paths, one for commands triggered with `!` and another for non-trigger text that _could_ match a regex?

Reply:
Would that mean that messages beginning '!' would also be permitted to trigger via a regex?
Could have implications for multiple matches, but seems like that could work as a refactor.